js/tests/unit/modal.js

```diff
@@ -627,27 +627,26 @@ $(function () {
 
   QUnit.test('should not follow link in area tag', function (assert) {
     assert.expect(2)
-    var done = assert.async()
 
     $('<map><area id="test" shape="default" data-toggle="modal" data-target="#modal-test" href="demo.html"/></map>')
       .appendTo('#qunit-fixture')
 
     $('<div id="modal-test"><div class="contents"><div id="close" data-dismiss="modal"/></div></div>')
       .appendTo('#qunit-fixture')
 
-    var evt = document.createEvent('HTMLEvents')
-    evt.initEvent('click', true, true)
+    // We need to use CustomEvent here to have a working preventDefault in IE tests.
+    var evt = new CustomEvent('click', {
+      bubbles: true,
+      cancelable: true
+    })
```

Comment:
that's why previously I asked you to use `EventHandler` in our unit test :wink:

Reply:
I've tried to use `EventHandler.trigger` instead of creating a `CustomEvent` object, but `preventDefault` produces the same buggy result. Trying to replace the `preventDefault` in an `HTMLEvents` object works on IE, but breaks in phantom. This is the only working solution for all the browsers.

Reply:
that's weird I didn't see that issue in our other plugins without jQuery

Reply:
I think this is the only test for the `defaultPrevented` property for a non-jQuery event by now. In other tests `trigger` is used to emit the event (which works correctly), but the Event object is discarded and never checked. 